docs/openapi.yaml

```diff
@@ -1,791 +1,1564 @@
+        - User
       requestBody:
-        required: true
-        description: A Comment object
-        content:
-          application/json:
-            schema:
-              $ref: '#/components/schemas/Comment'
+        $ref: '#/components/requestBodies/User_POST'
       responses:
         '201':
-          description: Successfully added new Comment
+          $ref: '#/components/responses/User_POST_201'
         '400':
-          description: Invalid request body
-          content:
-            application/json:
-              schema:
-                $ref: '#/components/schemas/Error'
+          $ref: '#/components/responses/400'
+        '401':
+          $ref: '#/components/responses/401'
         '403':
-          description: Request not made by an authorized user
-          content:
-            application/json:
-              schema:
-                $ref: '#/components/schemas/Error'
-        '404':
-          description: Plan with the specified `planId` not found
-          content:
-            application/json:
-              schema:
-                $ref: '#/components/schemas/Error'
+          $ref: '#/components/responses/403'
         '500':
-          description: Internal server error
-          content:
-            application/json:
-              schema:
-                $ref: '#/components/schemas/Error'
+          $ref: '#/components/responses/500'
 
+  '/user/authenticated':
+    get:
+      summary: Test the authentication of the current User
+      description: Tests the authentication of the current User.
+      operationId: authenticateUser
+      tags:
+        - User
+      responses:
+        '200':
+          $ref: '#/components/responses/User_Authenticated_GET_200'
+        '401':
+          $ref: '#/components/responses/401'
 
-  '/comment/{commentId}':
+  '/user/login':
     get:
-      summary: Fetch data about a specific Comment
+      summary: Verify the User's ticket from ONID CAS
       description: >
-        Returns information about a Comment with the provided `commentId`
-      operationId: getCommentById
+        Verifies the User's credentials after redirecting from CAS by sending
+        the ticket received from the redirection to the CAS service validator.
+      operationId: casValidateUser
       tags:
-        - Comment
+        - User
       parameters:
-        - name: commentId
-          in: path
-          description: ID of the Comment to fetch
-          required: true
-          schema:
-            type: integer
-            format: int64
-          example: 72
+        - $ref: '#/components/parameters/User_Login_GET_Ticket'
+        - $ref: '#/components/parameters/User_Login_GET_Target'
       responses:
         '200':
-          description: Success
+          $ref: '#/components/responses/User_Login_GET_200'
+        '401':
+          $ref: '#/components/responses/401'
+        '500':
+          $ref: '#/components/responses/500'
+
+  '/user/{userId}/plans':
```

Comment:
GET /user/{userId}/plans

shows this example value:

```
{
  "plans": [
    {
      "planId": 72,
      "status": 3,
      "planName": "My AI plan",
      "studentId": 72727272727,
      "created": "2020-12-31 23:59:59",
      "lastUpdated": "2020-12-31 23:59:59"
    }
  ]
}
```

However it should show advisor info also (since it needs to generate the advisors coins.
So something like the following is more appropriate:

```
{
  "plans": [
    {
      "planId": 72,
      "status": 3,
      "planName": "My AI plan",
      "studentId": 72727272727,
      "created": "2020-12-31 23:59:59",
      "lastUpdated": "2020-12-31 23:59:59"
      "advisors": [
        "John Doe", "Jane Doe"
      ]
    }
  ]
}
```

Reply:
Good catch. I was writing the specs with the old version of `master`. I'll update it.